kyaml/setters2/set.go

```diff
@@ -109,67 +109,100 @@ func (s *Set) visitScalar(object *yaml.RNode, p string, schema *openapi.Resource
 
 // substitute updates the value of field from ext if ext contains a substitution that
 // depends on a setter whose name matches s.Name.
-func (s *Set) substitute(field *yaml.RNode, ext *cliExtension, _ *spec.Schema) (bool, error) {
-	nameMatch := false
-
+func (s *Set) substitute(field *yaml.RNode, ext *cliExtension) (bool, error) {
 	// check partial setters to see if they contain the setter as part of a
 	// substitution
 	if ext.Substitution == nil {
 		return false, nil
 	}
 
-	p := ext.Substitution.Pattern
+	// track the visited nodes to detect cycles in nested substitutions
+	visited := sets.String{}
+
+	nameMatch := false
+
+	res, err := s.substituteUtil(ext, &visited, &nameMatch)
```

Comment:
`sets.String` is just a map, so there is not need to pass the reference here. We can just pass `visited` directly.

Reply:
Done.